File: Backend_dot_net_c_sharp_models_and_eums_to_convert_to_dart /Models/Placement/PlacementAnnouncement.cs
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
using PeerageHRIS.Constants;
using PeerageHRIS.Enumeration;
using PeerageHRIS.Enumeration.Training;

//using PeerageHRIS.Models.Commission;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace PeerageHRIS.Models.Placement
{
    public class PlacementAnnouncement
    {
        [Key]
        [Required]
        [Display(Name = "Announcement Id")]
        public int AnnouncementId { get; set; }

        [Required]
        [Display(Name = "Announcement Title")]
        public string? AnnouncementTitle { get; set; }

        [Required]
        [Display(Name = "Location")]
        public string? Location { get; set; }

        [Required]
        [Display(Name = "Department")]
        public string? Department { get; set; }

        [Required]
        [Display(Name ="Position")]
        public string? Position { get; set; }

        //[ForeignKey(nameof(PositionId))]
        //[JsonIgnore]
        //[ValidateNever]
        //public CommissionDesignation? Position {  get; set; }

        [Required]
        [Display(Name = "Required Experience (years)")]
        public int RequiredExperience { get; set; }

        [Required]
        [Display(Name = "Required Education")]
        public PlacementEducation? RequiredEducation { get; set; }

        [Required]
        [Display(Name ="Required Applicants")]
        public int RequiredApplicants { get; set; }

        [Display(Name = "Created Date")]
        [DataType(DataType.Date)]
        public DateTime? CreatedDate { get; set; } = PeerageConstant.CurrentTimeStamp.Date;

        [Display(Name = "Created By")]
        public string? CreatedBy { get; set; }

        [Display(Name = "Expiry Date")]
        [DataType(DataType.Date)]
        public DateTime? ExpiryDate { get; set; }

        [Display(Name ="Budgeted")]
        public bool Budgeted { get; set; } = true;

        [Display(Name ="Status")]
        public ApplicantStatus AnnouncementStatus { get; set; } = ApplicantStatus.Pending;


    }
}

File: lib/features/placement/domain/placement_announcement.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:police_com/core/enums/applicant_status.dart';
import 'package:police_com/core/enums/placement_education.dart';

part 'placement_announcement.freezed.dart';
part 'placement_announcement.g.dart';

@freezed
abstract class PlacementAnnouncement with _$PlacementAnnouncement {
  factory PlacementAnnouncement({
    required int announcementId,
    required String announcementTitle,
    required String location,
    required String department,
    required String positionId,
    required int requiredExperience,
    required PlacementEducation requiredEducation,
    required int requiredApplicants,
    required DateTime createdDate,
    required String createdBy,
    required DateTime expiryDate,
    required bool budgeted,
    required ApplicantStatus announcementStatus,
  }) = _PlacementAnnouncement;

  factory PlacementAnnouncement.fromJson(Map<String, dynamic> json) =>
      _$PlacementAnnouncementFromJson(json);
}
